Amp ooyala

chibzu opened the pull request.

jridgewell commented:
Fixes #4320.

jridgewell commented on extensions/amp-ooyala/0.1/amp-ooyala.js:
```diff
@@ -0,0 +1,75 @@
+  createPlaceholderCallback() {
+    const placeholderUrl = this.element.getAttribute("data-placeholder");
+    if (placeholderUrl) {
+      const placeholder = this.getWin().document.createElement('div');
```
Is the `<div>` necessary?

jridgewell commented on extensions/amp-ooyala/0.1/amp-ooyala.js:
```diff
@@ -0,0 +1,75 @@
+    let src = 'https://player.ooyala.com/iframe.html?platform=html5-priority';
+    let playerVersion = this.element.getAttribute('data-playerversion') || '';
+    if (playerVersion.toLowerCase() == 'v4') {
+      src = 'https://player.ooyala.com/static/v4/stable/latest/skin-plugin/amp_iframe.html?pcode=' + pCode;
```
You'll need to `encodeURIComponent` all these variables you retrieve from attributes.

rudygalfi commented:
How about `ooyala-player` for the component name?

jridgewell commented:
> How about ooyala-player for the component name?

Agreed, we've been giving all the video players a `-player` suffix (`jwplayer` got an exclusion, but that one's in the name).

chibzu commented:
No problem, will change it to AmpOoyalaPlayer and amp-ooyala-player.

chibzu commented:
I signed it!

chibzu commented:
Not sure why my tests aren't able to access user.assert from the '../../../src/log' import, any advice?

jridgewell commented:
We changed it to avoid a [static side-effect export](https://github.com/ampproject/amphtml/pull/4261). You'll have to update it to `user().assert(...)` now. Sorry about that.

chibzu commented:
Got past the user.assert issue with your advice, thanks! Still having trouble getting everything resolved with the validator. Is there any additional documentation for generating the validator .out files? Can't quite seem to get it right.

jridgewell commented:
@Gregable?

erwinmombay commented on examples/ooyalaplayer.amp.html:
```diff
@@ -0,0 +1,31 @@
+
```
can you remove first line? thanks!

mityaha replied:
@erwinmombay, I am not sure what are you meaning here...

erwinmombay commented on extensions/amp-ooyala-player/0.1/test/test-amp-ooyala.js:
```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
```
2016

erwinmombay commented on extensions/amp-ooyala-player/0.1/test/validator-amp-ooyala.html:
```diff
@@ -0,0 +1,75 @@
+<!--
+  Copyright 2015 The AMP HTML Authors. All Rights Reserved.
```
2016

erwinmombay commented on extensions/amp-ooyala-player/amp-ooyala-player.md:
```diff
@@ -0,0 +1,80 @@
+<!---
+Copyright 2015 The AMP HTML Authors. All Rights Reserved.
```
2016

erwinmombay commented:
think greg is out right now, ill try and help on the validator

erwinmombay commented on extensions/amp-ooyala-player/0.1/amp-ooyala-player.js:
```diff
@@ -0,0 +1,89 @@
+import {loadPromise} from '../../../src/event-helper';
+import {user} from '../../../src/log';
+
+class AmpOoyalaPlayer extends AMP.BaseElement {
```
@chibzu is there a pause mechanism you can implement?

chibzu replied:
There isn't any iframe messaging in place, no. What kind of behavior are you recommending?

mkhatib replied:
@chibzu for players it's recommended to implement a pausing mechanism so that AMP runtime can pause the video when the user navigates away of the document. For players that don't support a pausing mechanism the only option is to unlayout the player and layout it back again if the user navigates back again - the problem with this is it won't remember where the user left off.

If there's no way to implement pause in the player you need to override `unlayoutOnPause` to return `true`.

erwinmombay commented on extensions/amp-ooyala-player/0.1/amp-ooyala-player.js:
```diff
@@ -0,0 +1,89 @@
+
+  /** @override */
+  createPlaceholderCallback() {
+    const placeholderUrl = this.element.getAttribute('data-placeholder');
```
@mkhatib do you know if we're ok with this?

erwinmombay replied:
im going to say no, as is this might be requested on prerender

jridgewell replied:
Shouldn't this be done with a placeholder image inside the `<amp-ooyala>` tag?

mkhatib replied:
What @jridgewell said. This callback is only for generating dynamic placeholders. Users should already be able to do this  without you overriding this callback. Drop the entire callback.

If your service has a simple endpoint to retrieve thumbnails or poster for a video (e.g `https://thumbs.ooyala.com/${this.videoId_}/${this.otherParam_}/${this.width_}x${this.height_}`) you can then use this as your dynamic placeholder if the user haven't provided one using the [normal placeholder mechanism that AMP supports](https://github.com/ampproject/amphtml/blob/master/spec/amp-html-layout.md#placeholder).

See amp-instagram for an example usage.

erwinmombay commented:
@adding @mkhatib for a second pair of eyes, as he's done more video component review than i have

erwinmombay commented:
ping @mkhatib

jridgewell commented on extensions/amp-ooyala-player/0.1/amp-ooyala-player.js:
```diff
@@ -0,0 +1,89 @@
+      'allow-scripts allow-popups allow-same-origin');
+    iframe.src = src;
+    this.applyFillContent(iframe);
+    iframe.width = width;
```
These two lines are unnecessary. Re: https://github.com/ampproject/amphtml/pull/4674

mkhatib replied:
Also drop what would become unused width and height constants declaration at the top of the function.

FYI: Reason these are unnecessary is that you the `iframe` element will just fill the content of its parent element `amp-ooyala-player` because you called `applyFillContent`

mkhatib commented on examples/ooyalaplayer.amp.html:
```diff
@@ -0,0 +1,30 @@
+      data-playerid="26e2e3c1049c4e70ae08a242638b5c40">
+    </amp-ooyala-player>
+  </body>
+</html>
```
nit: new line

mkhatib commented on extensions/amp-ooyala-player/0.1/amp-ooyala-player.js:
```diff
@@ -0,0 +1,89 @@
+    this.element.appendChild(iframe);
+    return loadPromise(iframe);
+  }
+
```
Unless you can provide a pausing mechanism as discussed above, please also override and implement `unlayoutCallback`. In there your element needs to remove the iframe and return true. And `unlayoutOnPause` to return true. See amp-instagram for an example.

If you can provide pausing mechanism you need to override `pauseCallback`. See amp-brid-player for an example.

Gregable commented on extensions/amp-ooyala-player/0.1/validator-amp-ooyala-player.protoascii:
```diff
@@ -0,0 +1,83 @@
+  tag_name: "AMP-OOYALA-PLAYER"
+  also_requires_tag: "amp-ooyala-player extension .js script"
+  attrs: {
+    name: "data-embedcode"
```
would you like any attribute value constraints on these to help folks avoid typos? For example are data-embedcode or data-playerid only alphanumeric, only lower case, have a certain length, etc?

If not that's fine.

Gregable commented on extensions/amp-ooyala-player/0.1/validator-amp-ooyala-player.protoascii:
```diff
@@ -0,0 +1,83 @@
+    mandatory: true
+  }
+  attrs: {
+    name: "data-placeholder"
```
These can be left out, since all data-* attributes are allowed by default. If you want to leave them in as semi-documentation, that is OK too.

Gregable commented on extensions/amp-ooyala-player/0.1/validator-amp-ooyala-player.protoascii:
```diff
@@ -0,0 +1,83 @@
+    supported_layouts: FIXED
+    supported_layouts: FIXED_HEIGHT
+    supported_layouts: FLEX_ITEM
+    supported_layouts: NODISPLAY
```
does this layout type make sense?

Gregable commented on extensions/amp-ooyala-player/0.1/validator-amp-ooyala-player.protoascii:
```diff
@@ -0,0 +1,83 @@
+  amp_layout: {
+    supported_layouts: FILL
+    supported_layouts: FIXED
+    supported_layouts: FIXED_HEIGHT
```
does this layout type make sense? It'll result in changing the ratio of width and height.

Gregable commented on extensions/amp-ooyala-player/0.1/validator-amp-ooyala-player.protoascii:
```diff
@@ -0,0 +1,83 @@
+#
+
+tags: {  # amp-ooyala-player
+  tag_name: "SCRIPT"
```
please add to the top of this tag: section the following:
```
html_format: AMP
html_format: A4A
```

Gregable replied:
Oops, add this to the actual amp-gfycat tag, not the script tag.

mityaha replied:
Done

jridgewell commented on extensions/amp-ooyala-player/0.1/amp-ooyala-player.js:
```diff
@@ -0,0 +1,85 @@
+        encodeURIComponent(this.element.getAttribute('data-embedcode')),
+        'The data-embedcode attribute is required for <amp-ooyala-player> %s',
+        this.element);
+    const pCode = user().assert(
```
We're asserting there's a `data-pcode`, but it looks to only be necessary for v4 version. Is that intentional?

chibzu replied:
Yes, just keeping the usage relatively consistent between the two versions for simplicity. The three mandatory fields should allow any configuration of the player to work, current or future.

jridgewell commented on extensions/amp-ooyala-player/0.1/amp-ooyala-player.js:
```diff
@@ -0,0 +1,85 @@
+    this.iframe_ = iframe;
+
+    const embedCode = user().assert(
+        encodeURIComponent(this.element.getAttribute('data-embedcode')),
```
Please remove the `encodeURIComponent` here, you're going to double encode it (you already encode during the url generation below).

jridgewell commented on extensions/amp-ooyala-player/0.1/amp-ooyala-player.js:
```diff
@@ -0,0 +1,85 @@
+  /** @override */
+  unlayoutCallback() {
+    if (this.iframe_) {
+      removeElement(this.iframe_);
```
You'll need to import `removeElement` from the `dom.js` module.

chibzu replied:
Added in 87a4e6db37a7638be10b34d5bd1204a6bca11133

erwinmombay commented:
ping @chibzu , would love to get this going again. let us know how we can help or if you had any questions

chibzu commented:
Been addressing the comments when I've had time, I think I've addressed most of them by now. It looks like I broke some validation in adding the "html_format: A4A" in the protoascii file in commit 95864ddd23b2a2f4f0e5fc0e804cbb03543132f6 . Should be be A4AAmp or did I put it in the wrong place?
Thanks for all the feedback so far, I appreciate it.

chibzu commented:
I think there's just one more thing failing in the travis job, which has to do with the DISALLOWED_TAG in test-example-validation.  Should I remove amp-ooyala-player from that array or is it necessary?

chibzu commented:
Sorry to be a bother, but I'm not sure what the next steps are for this. Any guidance is appreciated.

jridgewell commented:
It seems we have some validation issues. @Gregable?

Gregable commented:
Please change `html_format: A4A` to `html_format: AMP4ADS` in `validator-amp-ooyala-player.protoascii`.
The naming changed during this PR, so it's a bit of a moving target. Should be a quick fix though. Apologies for the hassle on this, and that I was not following the thread closely enough.

Otherwise the PR looks good.

Gregable commented:
The validator changes look good to me. @mkhatib , @erwinmombay is this ready to go?

chibzu commented:
Very exciting, thanks for all the feedback so far!

erwinmombay commented on test/integration/test-example-validation.js:
```diff
@@ -55,6 +55,7 @@ describe.configure().retryOnSaucelabs().run('example', function() {
     'facebook.amp.html',
     'gfycat.amp.html',
     'instagram.amp.html',
+    'ooyala.amp.html',
```
this file doesnt exist right? think you might need to rebase from master, i tried to figure out how to fix your test issues and there were some conflicts for this file. let me know when you've rebased then we can try and figure out how to fix the issues.

think you can just reset test-example-validation.js
and then resolve gulpfile.js as well, (might need to reset and then just re-add the amp-ooyala entry)

we're close!

chibzu commented:
I merged with upstream master and changed the filename. There weren't any conflicts and it looks up to date. Let me know if it still needs a rebasing.

erwinmombay commented on examples/ooyalaplayer.amp.html:
```diff
@@ -0,0 +1,30 @@
+    <link rel="canonical" href="amps.html" />
+    <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+    <style>body {opacity: 0}</style><noscript><style>body {opacity: 1}</style></noscript>
+    <script async custom-element="amp-ooyala-player" src="https://cdn.ampproject.org/v0/amp-ooyala-player-0.1.js"></script>
```
please change this to `<script async custom-element="amp-ooyala-player" data-amp-report-test="amp-ooyala-player" src="https://cdn.ampproject.org/v0/amp-ooyala-player-0.1.js"></script>`

we unfortunately need the `data-amp-report-test` for testing

erwinmombay commented on test/integration/test-example-validation.js:
```diff
@@ -75,6 +76,7 @@ describe.configure().retryOnSaucelabs().run('example', function() {
   const errorWhitelist = [
     /GENERAL_DISALLOWED_TAG script viewer-integr.js/,
     /DISALLOWED_TAG content/,  // Experiments with shadow slots
+    /DISALLOWED_TAG amp-ooyala-player/,
```
also add `/GENERAL_DISALLOWED_TAG script amp-ooyala-player/,`

mityaha replied:
Done

erwinmombay commented:
just need those last 2 changes and the tests/build should be green

erwinmombay commented:
@aghassemi can you do a pass on this? as you're familiar with these video components than I am. Thanks!

erwinmombay commented:
@aghassemi, @mkhatib also gave this a review pass too, so even a quick one would be appreciated. Thanks!

aghassemi commented on extensions/amp-ooyala-player/0.1/amp-ooyala-player.js:
```diff
@@ -0,0 +1,96 @@
+import {user} from '../../../src/log';
+import {removeElement} from '../../../src/dom';
+
+class AmpOoyalaPlayer extends AMP.BaseElement {
```
We now expect all new video-players to implement our video-interface. 

- Please see https://github.com/ampproject/amphtml/blob/master/src/video-interface.js for the methods, attributes and events players are expected to implement.
- Please see https://github.com/ampproject/amphtml/blob/master/builtins/amp-video.js for an example of a video player that fully implements this interface.
- Please see https://github.com/aghassemi/amphtml/blob/video-yt/extensions/amp-youtube/0.1/amp-youtube.js for an iframe-based player that implements this interface (Note this code is WIP and might be out of date)

mityaha replied:
Does it mean that we should implement: supportsPlatform, showControls, hideControls, element, isInViewPort?  I don't see these implementations by other extensions

aghassemi replied:
We are slowly going through existing players and adding support. `amp-youtube` currently is the canonical example that implements most of the API. 

As mentioned in previous comment, required methods are `play`, `pause`, `mute`, `unmute`. Just return `true` for `supportsPlatform` and `isInteractive` and leave implementation for `showControls` and `hideControls` empty.

mityaha replied:
OK, understood.  Done.

aghassemi commented on extensions/amp-ooyala-player/0.1/amp-ooyala-player.js:
```diff
@@ -0,0 +1,96 @@
+      '&pbid=' + encodeURIComponent(playerId);
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowfullscreen', 'true');
+    iframe.setAttribute('sandbox',
```
Remove setting sandbox. `allow-scripts + allow-same-origin` essentially makes `sandbox` pointless anyway.

mityaha replied:
It is not clear to me why setting sandbox has to be removed.  According to [documentation](https://github.com/ampproject/amphtml/blob/master/extensions/amp-iframe/amp-iframe.md#sandbox ) setting sandbox="allow-scripts" allows the iframe to run JavaScript, or sandbox="allow-scripts allow-same-origin" allows the iframe to run JavaScript, make non-CORS XHRs, and read/write cookies.

mityaha replied:
I see, thanks for clarifying.  Removed.

aghassemi commented on extensions/amp-ooyala-player/0.1/amp-ooyala-player.js:
```diff
@@ -0,0 +1,96 @@
+
+  /** @override */
+  unlayoutOnPause() {
+    return true;
```
Since part of the new expected interface requires you to implement a `pause` method anyway, I recommend overriding `pauseCallback` and setting this to `false` so when AMP document visibility changes, we can simply pause the video instead of destroying it and losing state.

chibzu commented:
Are these going to be hard requirements for this pull request to be accepted? Our player doesn't currently have messaging through the iframe boundary, which would need to be added to our hosted iframe. Another bigger issue is the "controls" attribute as our player doesn't have a toggle for completely disabling the UI. This would require another round of development and deployment on our end which is tough with the integration being a bit of a moving target as it is right now. Is there an expected timeline for when the integration points are going to be cemented into a 1.0 type of state?

aghassemi commented:
@chibzu We are flexible on not having `controls` part implemented but we like to have `play()`, `pause()`, `mute()` and `unmute()` implemented before merging this PR. That part of the API is pretty solid and would enable muted autoplay for your player. (autoplay will be fully managed by AMP runtime, all you need to do is to implement those four methods and ensure your HTML5 `video` tag has `playsinline` attribute on it for iOS). Since the `USER_TAP` event is tied to functionality related to controls, you can skip that event as well for now.

chibzu commented:
Got it, thanks. Will see what we can get done about those control methods.

aghassemi commented:
@chibzu Let me know when this is ready for review again. We can merge when `pause`, `resume`, `mute` and `unmute` methods are implemented and work from the common API side of things.

aghassemi commented:
@chibzu `<amp-youtube>` now supports the video-interface and we have a test suite to run against players. `<amp-youtube>` is using `postMessage` to implement the interface.

Please look at the changes to `amp-youtube.js` here https://github.com/ampproject/amphtml/commit/1a1bddedd948ca9713dd5ccf4ff3c96847414a36#diff-6902ec55f2eb8f53354fe4ed9eb1e28bL22 for reference.

Adding compliance test is as simple as adding a few lines in `test/integration/test-video-players.js` creating an instance of your player. The rest is handled by test suite which would test methods such as `pause()/play()` and the expected events such as `pause/play`.

As mentioned earlier, implementing `show/hide controls` methods are optional. `<amp-youtube>`  does not implement them either.

mityaha commented:
I signed it!

mityaha commented on extensions/amp-ooyala-player/amp-ooyala-player.md:
````diff
@@ -0,0 +1,76 @@
+```
+
+## Attributes
+
````
I don't see actual mentions and implementations of autoplay and controls parameters by instagram, dailymotion, facebook, jwplayer, brightcove, etc.  I am only seeing mentioning of data-params-controls in amp-youtube.md.  Is it really required for merge of our first implementation?

mityaha replied:
OK, clear.  Is there anything else that has to be implemented / reworked/ fixed in this pull request?

mityaha commented:
I signed it!

> On Nov 29, 2016, at 13:14, googlebot <notifications@github.com> wrote:
> 
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 Please visit https://cla.developers.google.com/ <https://cla.developers.google.com/> to sign.
> 
> Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.
> 
> If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address. Check your existing CLA data <https://cla.developers.google.com/clas> and verify that your email is set on your git commits <https://help.github.com/articles/setting-your-email-in-git/>.
> If you signed the CLA as a corporation, please let us know the company's name.
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/ampproject/amphtml/pull/4333#issuecomment-263700708>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AM7zjuyADFlELnF3n6DE9r02v8Sj4xEyks5rDJWlgaJpZM4JbH_h>.
>

aghassemi commented on extensions/amp-ooyala-player/0.1/amp-ooyala-player.js:
```diff
@@ -0,0 +1,139 @@
+    iframe.setAttribute('allowfullscreen', 'true');
+    iframe.src = src;
+    this.applyFillContent(iframe);
+
```
nit: remove extra new line

mityaha replied:
done

aghassemi commented on extensions/amp-ooyala-player/0.1/amp-ooyala-player.js:
```diff
@@ -0,0 +1,139 @@
+    this.applyFillContent(iframe);
+
+    this.element.appendChild(iframe);
+    return this.loadPromise(iframe);
```
we need to dispatch a `VideoEvents.LOAD` event here, so change this to:

```
      return this.loadPromise(iframe).then(() => {
        this.element.dispatchCustomEvent(VideoEvents.LOAD);
      });
```

mityaha replied:
done

aghassemi commented on extensions/amp-ooyala-player/0.1/amp-ooyala-player.js:
```diff
@@ -0,0 +1,139 @@
+  }
+
+  /** @override */
+  isInViewport() {
```
`isInteractive` not `isInViewport`

mityaha replied:
Are you sure? VideoInterface has isInViewPort and it doesn't have isInteractive.  Actually, there is no isInteractive strings at all in the source code.

mityaha replied:
done.  Will rebase

mityaha replied:
I have rebased this branch with master (git rebase master) and there were some merge conflicts in gulpfile.js and test-example-validation.js. I resolved conflicts by re-applying just our changes to master versions of these files, checked them in, but somehow a lot of old code in these files is still seen in this pull request. Then I deleted local copy of amphtml and re-checked out amp-ooyala branch and I now I am seeing that this branch is gone!  
Fortunately, I had all copies of files in my editor and I created a new amp-oolaya branch (it allowed me to do git checkout -b amp-ooyala) and restored all our code and modifications to other files.  But now I can't push this branch as I don't have permissions:
git push -u origin amp-ooyala
remote: Permission to ampproject/amphtml.git denied to mityaha.
fatal: unable to access 'https://github.com/ampproject/amphtml.git/': The requested URL returned error: 403

What did I do wrong? What was I supposed to rebase our branch with?  Why did our amp-ooyala branch go away? What should I do now?  Please help!

Also, on each commit I was performing, I was receiving email from googlebot, "Thanks for your pull request. It looks like this may be your first contribution to a Google open source project. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA)."  I did sign agreement, I did reply with "I signed it!" message, but I keep receiving these responses from googlebot.  How can this problem be fixed?

aghassemi commented on extensions/amp-ooyala-player/0.1/amp-ooyala-player.js:
```diff
@@ -0,0 +1,139 @@
+  unlayoutOnPause() {
+    return false;
+  }
+
```
add
```
// VideoInterface Implementation. See ../src/video-interface.VideoInterface
```


and move `pause`, `play`, `mute`, `unmute` methods below this line

mityaha replied:
done

aghassemi commented on extensions/amp-ooyala-player/0.1/validator-amp-ooyala-player.protoascii:
```diff
@@ -0,0 +1,80 @@
+  attrs: {
+    name: "data-config"
+  }
+  attr_lists: "extended-amp-global"
```
add:

```
  attrs: { 
    name: "autoplay" 
  }
```

mityaha replied:
done

aghassemi commented on extensions/amp-ooyala-player/amp-ooyala-player.md:
```diff
@@ -0,0 +1,76 @@
+**data-config**
+
+An optional attribute to specify a skin.json config file URL for player V4.
+
```
add:

```
**autoplay**

If this attribute is present, and the browser supports autoplay:

* the video is automatically muted before autoplay starts
* when the video is scrolled out of view, the video is paused
* when the video is scrolled into view, the video resumes playback
* when the user taps the video, the video is unmuted
* if the user has interacted with the video (e.g., mutes/unmutes, pauses/resumes, etc.), and the video is scrolled in or out of view, the state of the video remains as how the user left it.  For example, if the user pauses the video, then scrolls the video out of view and returns to the video, the video is still paused. 
```

mityaha replied:
done

mityaha commented on extensions/amp-ooyala-player/0.1/amp-ooyala-player.js:
```diff
@@ -0,0 +1,139 @@
+    /** @private {?Element} */
+    this.iframe_ = null;
+  }
+
```
done

aghassemi commented:
@mityaha You were getting the permission denied, because you were pushing your branch to `ampproject/amphhtml` you need to push to your fork instead (`ooyala/amphtml`).

For the CLA, please ensure the email address used for your commits matches the one you registered:
If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address.  Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).

aghassemi commented:
@mityaha (btw, feel free to send a new PR if the Git stuff is too tangled now)

mityaha commented:
@aghassemi, thanks for clarifying about ooyala/amphtml fork, I wasn't aware of it.  Now I am able to get back code from amp-ooyala branch.  

I reset user.email to an alias that I was using in my git account, hopefully it will eliminate CLA problem.

I followed up on all your comments that you inlined in the diffs page.  I haven't yet done steps that you mentioned separately (about VideoEvents.PAUSE, VideoEvents.PLAYING) as I want to settle gulpfile problems first.

I am not sure what to do with gulpfile.js.  I took version from master (from ooyala/amphtml fork) and added a line about ooyala amp player.  Checked it in, but diff shows tons of changes.  What did I do wrong or what should I do?

You also said I should rebase.  What should I rebase with?  Yesterday I did 'git rebase master' and it created a few conflicts with gulpfile.js and test-example-validation.js.  I fixed problems with the latter, but gulpfile.js is still messed up.

aghassemi commented:
@mityaha I suggest just resetting the gulpfile.js (`git checkout upstream/master gulpfile.js`) and adding the ooyala code again.

I am also suggesting to pull latest amphtml code from the upstream (i.e. amphtml) to your branch. ( `git pull upstream master` ) If you don't have upstream setup, you can add it like this: `git remote add upstream https://github.com/ampproject/amphtml.git`

aghassemi commented on gulpfile.js:
```diff
@@ -85,6 +85,7 @@ declareExtension('amp-list', '0.1', false, 'NO_TYPE_CHECK');
 declareExtension('amp-live-list', '0.1', true);
 declareExtension('amp-mustache', '0.1', false, 'NO_TYPE_CHECK');
 declareExtension('amp-o2-player', '0.1', false, 'NO_TYPE_CHECK');
+declareExtension('amp-ooyala-player', '0.1', false, 'NO_TYPE_CHECK');
```
please enable type check by removing 'NO_TYPE_CHECK' and running `gulp check-type` to ensure all is good.

mityaha commented:
@aghassemi, thanks for your latest recommendations, I was able to bring sources back.  

I am still facing a problem with this CLA thing.  My *real* address is dimitry.nasledov@ooyala.com, but I have an alias dima@ooyala.com.  I created github account a while ago with this alias.  I tried git config --global user.email with both of these addresses and it didn't work with either.

When I run gulp test --integration, I have these errors (see below) - what else am I missing?
[17:32:01] All current JS updates done.
Chrome 54.0.2840 (Mac OS X 10.12.1) amp-ooyala-player Autoplay play/pause should play when in view port initially FAILED
	Error: timeout of 20000ms exceeded. Ensure the done() callback is being called in this test.
Chrome 54.0.2840 (Mac OS X 10.12.1) amp-ooyala-player Autoplay play/pause should play/pause when video enters/exists viewport FAILED
	Error: timeout of 20000ms exceeded. Ensure the done() callback is being called in this test.
Chrome 54.0.2840 (Mac OS X 10.12.1) amp-ooyala-player Autoplay Animated Icon should create an animated icon overlay FAILED
	Error: Timeout waiting for animation icon
	    at poll (/var/folders/qh/3x5wp1550k53zh04fplqwhsc0000gq/T/1579654ead4645c84620388791f1ea12.browserify:42553:18 <- /Users/dimitry.nasledov/repos/amphtml/testing/iframe.js:381:17)
Chrome 54.0.2840 (Mac OS X 10.12.1): Executed 73 of 95 (3 FAILED) (skipped 22) (49.223 secs / 45.701 secs)
[17:32:52] Shutting down test responses server on localhost:31862
[17:32:52] 'test' errored after 2.53 min

Also, as you recommended, I removed NO_TYPE_CHECK, and ran the test.  Here is the output:

 gulp check-type
[17:38:30] Using gulpfile ~/repos/amphtml/gulpfile.js
[17:38:30] Task 'check-type' is not in your gulpfile
[17:38:30] Please check the documentation for proper gulpfile formatting

mityaha commented:
I signed it!

-- 
Dimitry Nasledov
dima@ooyala.com


> On Nov 30, 2016, at 5:34 PM, googlebot <notifications@github.com> wrote:
> 
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 Please visit https://cla.developers.google.com/ to sign.
> 
> Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.
> 
> If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address. Check your existing CLA data and verify that your email is set on your git commits.
> If you signed the CLA as a corporation, please let us know the company's name.
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

mityaha commented:
I signed it!

-- 
Dimitry Nasledov
dima@ooyala.com


> On Nov 30, 2016, at 1:11 PM, googlebot <notifications@github.com> wrote:
> 
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 Please visit https://cla.developers.google.com/ to sign.
> 
> Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.
> 
> If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address. Check your existing CLA data and verify that your email is set on your git commits.
> If you signed the CLA as a corporation, please let us know the company's name.
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

aghassemi commented:
@mityaha PR is looking pretty good, just couple of last-mile stuff:

1- For CLA, please visit https://cla.developers.google.com/clas and use `Edit Contact Information` and see what email address and GitHub username show up there. I **think** if your GitHub username is set there, email should not matter.

2- For `gulp check-type` errors, my fault `command is `gulp check-types`

3- For failing test, as mentioned in https://github.com/ampproject/amphtml/pull/4333#pullrequestreview-10666241, we need the component to dispatch  `VideoEvents.PAUSE` and `VideoEvents.PLAYING` events when the underlying videoplayer is paused or played. This requires some sort of event messages coming from the iframed-player through `postMessage API`. See amp-youtube.js for how that is implemented. 

I recommend testing manually first to make sure the video-interface is implemented currently. You can do this by adding `autoplay` to your example page and ensure 
1- It autoplays on load, 
2- It pauses when scrolling causes it to go out of view,
3- It plays again when scrolling brings it back to view 
4- The animated equalizer icon that's overlayed on your player by AMP is also paused/played when video is paused/played as result of scrolling. 5- ensure tapping/clicking on the autoplaing video "unmutes" it.

mityaha commented:
@aghassemi  regarding CLA - yes, I went to this page a few times, my email address there is set to dimitry.nasledov@ooyala.com, my user name is 'mityaha' - see attached. I tried `git config --global user.email "dimitry.nasledov@ooyala.com"` - but it keeps sending these alerts.  
![screen shot 2016-12-01 at 11 35 06 am](https://cloud.githubusercontent.com/assets/13562766/20809378/b407ed9c-b7ba-11e6-9a8f-929ca048ef0b.png)

aghassemi commented:
@mityaha sorry about the CLA mess up, I can see you have signed the CLA internally. Something is going on with the bot. I am following up internally to fix this.

aghassemi commented:
@mityaha Because of a bad rebase, there are 250+ unrelated commits in this PR and that seems to be the reason the CLA check is failing ( also it is not correct to show 250 commits on this PR )

Could we please open a new pull request? I have already created a patch for you with the changes in this PR, what you can do is:
(The following assumes `origin` is `ooyala/amphtml` and `upstream` is `ampproject/amphtml`, update the remote names if your setup is different)

1- create a new named branch based on latest master:
`git checkout upstream/master -b amp-ooyala2`
2- apply the attached patch file [amp-ooyala2.diff.txt](https://github.com/ampproject/amphtml/files/625864/amp-ooyala2.diff.txt)
`git apply path/to/amp-ooyala2.diff`
3- push the name branch to origin and create a new PR


Thanks!
-Ali

aghassemi commented:
Closing in favour of a fresh PR at https://github.com/ampproject/amphtml/pull/6466
